app/src/test/java/com/eventyay/organizer/core/sessions/create/CreateSessionViewModelTest.java

```diff
@@ -0,0 +1,235 @@
+package com.eventyay.organizer.core.sessions.create;
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule;
+import android.arch.lifecycle.Observer;
+import com.eventyay.organizer.common.ContextManager;
+import com.eventyay.organizer.core.session.create.CreateSessionViewModel;
+import com.eventyay.organizer.data.event.Event;
+import com.eventyay.organizer.data.session.Session;
+import com.eventyay.organizer.data.session.SessionRepository;
+import com.eventyay.organizer.utils.DateUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+import org.threeten.bp.LocalDateTime;
+
+import io.reactivex.Observable;
+import io.reactivex.android.plugins.RxAndroidPlugins;
+import io.reactivex.plugins.RxJavaPlugins;
+import io.reactivex.schedulers.Schedulers;
+
+import static org.junit.Assert.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(JUnit4.class)
+public class CreateSessionViewModelTest {
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+    @Rule
+    public TestRule rule = new InstantTaskExecutorRule();
+    @Mock
+    private SessionRepository sessionRepository;
+    @Mock
+    private Event event;
+
+    @Mock
+    Observer<String> error;
+    @Mock
+    Observer<Boolean> progress;
+    @Mock
+    Observer<String> success;
+    @Mock
+    Observer<Void> dismiss;
+
+    private CreateSessionViewModel createSessionViewModel;
+    private static final Session SESSION = Session.builder().id(2L).title("dd").build();
+    private static final String ERROR = "Error";
+    private static final long EVENT_ID = 5L;
+    private static final long TRACK_ID = 5L;
+
+    @Before
+    public void setUp() {
+        RxJavaPlugins.setIoSchedulerHandler(scheduler -> Schedulers.trampoline());
+        RxAndroidPlugins.setInitMainThreadSchedulerHandler(schedulerCallable -> Schedulers.trampoline());
+
+        createSessionViewModel = new CreateSessionViewModel(sessionRepository);
+        ContextManager.setSelectedEvent(event);
+    }
+
+    @After
+    public void tearDown() {
+        RxJavaPlugins.reset();
+        RxAndroidPlugins.reset();
+    }
+
+    @Test
+    public void shouldRejectWrongDates() {
+        Session session = createSessionViewModel.getSession();
+
```

Comment:
Here you arn't setting a selected event in ContextUtils before checking for the correct date, this would make ZoneId null. Use `ContextManager.setSelectedEvent(event)`